### PR TITLE
Add profiler flag

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- https://github.com/obscuronet/obscuro-internal/issues/607
- Helpful to debug testnet issue

### What changes were made as part of this PR:

- Host and Enclave Flags

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -89,6 +92,15 @@ func NewEnclave(
 
 	nodeShortID := common.ShortAddress(config.HostID)
 
+	var prof *profiler.Profiler
+	if config.ProfilerEnabled {
```
Comment (tudor-malene) [inline go/enclave/enclave.go:96]: this should not be possible in secure mode.

**Review by tudor-malene — COMMENTED** [review]
didn't have time to look properly.
But I have one comment. We need to make sure that in "secure" mode you can't turn on the profiler and leak stuff from the enclave

File: go/enclave/enclave.go (hunk 1 above)
Comment (otherview) [reply go/enclave/enclave.go:96]: Good point - wonder if we should leave it to some sort of debug build ?

Maybe something to reiterate in the future? 

Removing the profiler from the enclave 👍

Comment (otherview) [reply go/enclave/enclave.go:96]: Update: Will tie this to the secure execution of the enclave.

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**